example-app/app/auth/effects/auth.effects.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { TestBed } from '@angular/core/testing';
+import { Actions } from '@ngrx/effects';
+import { cold, hot } from 'jasmine-marbles';
+import { Observable } from 'rxjs/Observable';
+import { empty } from 'rxjs/observable/empty';
+import { Router } from '@angular/router';
+
+import { AuthEffects } from './auth.effects';
+import { AuthService } from '../services/auth.service';
+import { Login, LoginFailure, LoginSuccess } from '../actions/auth';
+import { Authenticate, User } from '../models/user';
+
+export class TestActions extends Actions {
+  constructor() {
+    super(empty());
+  }
+
+  set stream(source: Observable<any>) {
+    this.source = source;
+  }
+}
+
+export function getActions() {
+  return new TestActions();
+}
+
+describe('AuthEffects', () => {
+  let effects: AuthEffects;
+  let authService: any;
+  let actions$: TestActions;
+  let routerService: any;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        AuthEffects,
+        {
+          provide: AuthService,
+          useValue: { login: jest.fn() },
+        },
+        {
+          provide: Actions,
+          useFactory: getActions,
+        },
+        {
+          provide: Router,
+          useValue: { navigate: jest.fn() },
+        },
+      ],
+    });
+
+    effects = TestBed.get(AuthEffects);
+    authService = TestBed.get(AuthService);
+    actions$ = TestBed.get(Actions);
+    routerService = TestBed.get(Router);
+
+    spyOn(routerService, 'navigate').and.callThrough();
+  });
+
+  describe('login$', () => {
+    it('should return an auth.LoginSuccess action, with user information if login succeeds', () => {
+      const credentials: Authenticate = { username: 'test', password: '' };
+      const user = { name: 'User' } as User;
+      const action = new Login(credentials);
+      const completion = new LoginSuccess({ user });
+
+      actions$.stream = hot('-a---', { a: action });
+      const response = cold('-a|', { a: user });
+      const expected = cold('--b', { b: completion });
+      authService.login = jest.fn(() => response);
+
+      expect(effects.login$).toBeObservable(expected);
+    });
+
+    it('should return a new auth.LoginFailure if the login service throws', () => {
+      const credentials: Authenticate = { username: 'someOne', password: '' };
+      const action = new Login(credentials);
+      const completion = new LoginFailure('Invalid username or password');
+      const error = 'Invalid username or password';
+
+      actions$.stream = hot('-a---', { a: action });
+      const response = cold('-#|', {}, error);
+      const expected = cold('--b', { b: completion });
+      authService.login = jest.fn(() => response);
+
+      expect(effects.login$).toBeObservable(expected);
+    });
+  });
+
```

Comment:
The two test below don't seem correct. You're not dispatching any actions to trigger login success and `router.navigate` should be called with the string wrapped in an array (e.g. `['/']`)

Reply:
Thank you for the comment!
I dispatch now the action and have verified that `router.navigate` method was indeed called with the right arguments (using `router.navigate.calls.allArgs()`). So no _false positive_ anymore 🙂